src/client/pages/AssessmentSection/Descriptions/Description/DataSources/DataSourceColumn/ColumnFraVariable.tsx

```diff
@@ -24,23 +24,37 @@ const ColumnFraVariable: React.FC<Props> = (props: Props) => {
   const _allColumnsCalculated = (row: Row) =>
     row.cols.every((col) => [ColType.header, ColType.calculated].includes(col.props.colType))
 
-  const rows = table.rows.reduce((acc, row) => {
+  const rows = table.rows.reduce<Record<string, string>>((acc, row) => {
     if (row.props.variableName && row.props.type === RowType.data && !_allColumnsCalculated(row)) {
       const label = t(Cols.getLabel({ cycle, col: row.cols[0], t }))
-      if (!acc.includes(label)) {
-        acc.push(label)
+      if (!acc[label]) {
+        // eslint-disable-next-line no-param-reassign
+        acc[label] = row.props.variableName
       }
     }
     return acc
-  }, dataSourceVariables?.include.map<string>(t) ?? [])
+  }, dataSourceVariables?.include.reduce((acc, variable) => ({ ...acc, [t(variable)]: variable }), {}) ?? {})
 
-  const _onChange = (value: any) => {
-    onChange('fraVariables', value)
+  const _onChange = (value: string[]) => {
+    onChange(
+      'fraVariables',
+      value.map((v) => rows[v])
+    )
   }
 
   return (
     <DataColumn className="data-source-column">
-      <MultiSelect disabled={disabled} values={dataSource.fraVariables ?? []} options={rows} onChange={_onChange} />
+      <MultiSelect
+        disabled={disabled}
+        values={Object.entries(rows).reduce((acc, [label, variable]) => {
```

Comment:
can't the rows be directly an array created by the reducer?

Reply:
it was the initial implementation, but the MultiSelect component expects an list of string

https://github.com/openforis/fra-platform/blob/2c7ca8604c63b586e77cf783702400d5e71f1353/src/client/components/MultiSelect/MultiSelect.tsx#L18-L19

We cannot translate in the MultiSelect component by :
https://github.com/openforis/fra-platform/blob/2c7ca8604c63b586e77cf783702400d5e71f1353/src/client/components/MultiSelect/MultiSelect.tsx#L31-L34

...because we have multiple prefixes:

https://github.com/openforis/fra-platform/blob/2c7ca8604c63b586e77cf783702400d5e71f1353/src/client/pages/AssessmentSection/Descriptions/Description/DataSources/DataSourceColumn/ColumnFraVariable.tsx#L27-L36


**dataSourceVariables?.include** contains a list of values to include (eg. "other, specify in comments")